src/main/java/com/aws/greengrass/clientdevices/auth/iot/registry/InMemoryRepository.java

```diff
@@ -0,0 +1,44 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package com.aws.greengrass.clientdevices.auth.iot.registry;
+
+
+import java.util.Collections;
+import java.util.LinkedHashMap;
+import java.util.Map;
+
+/**
+ * Mediates between the registry domain and the in-memory data storage.
+ *
+ * @param <K> Registry key type
+ * @param <V> Registry value type
+ */
+public class InMemoryRepository<K, V> implements Repository<K, V> {
```

Comment:
This is close but the Registry above should be the one implementing this interface